federation-js/src/composition/compose.ts

```diff
@@ -129,6 +129,11 @@ type FieldDirectivesMap = Map<string, DirectiveNode[]>;
 // TODO: rename?
 type TypeNameToFieldDirectivesMap = Map<string, FieldDirectivesMap>;
 
+/**
+ * A set of directive names that have been used at least once
+ */
+type AppliedDirectiveUsages = Set<string>;
```

Comment:
This is not a blocker, but the term 'applied directive usages' is a bit weird. Both 'applied directives' or 'directive usages' are meant to differentiate between directive definitions and their use.